jekyll/_cci2/api-developers-guide.md

```diff
@@ -0,0 +1,563 @@
+
+* 500 - Internal Server Error
+
+# REST API Tutorial
+
+For more detailed information about HTTP status codes, refer to the [rest api tutorial](http://www.restapitutorial.com/httpstatuscodes.html).
+
+# Example End-to-End API Request
+
+The following section details the steps you would need, from start to finish, to make an API call. This section includes creating a "demo repository" called "hello-world", however, you can use a pre-existing repository to follow along if you choose.
+
+## Prerequisites
+
+* A GitHub or BitBucket account with a repository to setup with CircleCI.
+* Completion of the CircleCI onboarding.
+* **Optional**: You can format JSON responses by piping the `curl` command into the `jq` utility if you have it installed: `curl ... | jq`.
+
+## Steps
+
+1. On your VCS provider, create a repository. The repo for this example will be called `hello-world`.
+
+2. Next, follow the onboarding for a new project on CircleCI. You can access onboarding by visiting the application and clicking on "Add Projects" in the sidebar or by going to the link: https://onboarding.circleci.com/project-dashboard/{VCS}/{ORG_NAME} where `VCS` is either `github` (or `gh`) or `bitbucket` (or `bb`) and `ORG_NAME` is your organization or personal VCS username. Find your project in the onboarding list and click `Setup Project`. After completing an onboarding, you should have a valid `config.yml` file in a `.circleci` folder at the root of your repository. In this example, the `config.yml` contains the following:
+
+    ```yaml
+    # Use the latest 2.1 version of CircleCI pipeline process engine. See: https://circleci.com/docs/2.0/configuration-reference
+    version: 2.1
+    # Use a package of configuration called an orb.
+    orbs:
+    # Declare a dependency on the welcome-orb
+      welcome: circleci/welcome-orb@0.4.1
+    # Orchestrate or schedule a set of jobs
+      workflows:
+      # Name the workflow "welcome"
+      welcome:
+      # Run the welcome/run job in its own container
+        jobs:
+        - welcome/run
+    ```
+
+3. Add an API token from your [account dashboard](https://circleci.com/account/api). Be sure to write down and store your API token in a secure place once you generate it.
+
+4. It's time to test out your API token using `curl` to make sure everything works. The following code snippets demonstrate querying all pipelines on a project. Please note that in the example below, the values within curly braces (`{}`) need to be replaced with values specific to your username/orgname.
+
+    ```sh
+    # First: set your CircleCI token as an environment variable
+    export CIRCLECI_TOKEN={your_api_token}
+        
+    curl --header "Circle-Token: $CIRCLECI_TOKEN" \
+      --header 'Accept: application/json'    \
+      --header 'Content-Type: application/json' \
+      https://circleci.com/api/v2/project/gh/{USER_NAME}/hello-world/pipeline 
```

Comment:
Replace 'gh' with {VCS} or {VCS-Provider}. 

All of these should be valid in 2.0:
gh
github
bb
bitbucket

Reply:
Fixed!